Added support for blank columns, empty typed rows and pointers

leonsteinhaeuser opened the pull request:
This PR adds support for empty columns and rows (rows only for typed data)

leonsteinhaeuser committed:
add: added support for blank columns and empty type rows

leonsteinhaeuser committed:
enhancement: added better support for empty rows, columns and comments

martinrode committed:
FIxed test to check json result properly

martinrode committed:
Added test for ptr type (fails)

leonsteinhaeuser committed:
enhancement: added pointer support

See: #2
See: #3

martinrode reviewed (changes requested):
* Initialize CSV with `CSVParser{}`.
* Make all options capitalized.  
* Sort all exported function to the top
* export "Typed()", "Untyped()" (renamed from ToTypedMap, ToUntypedMap)

leonsteinhaeuser committed:
enhancement: simplified the api

Introduced the requested changes from PR: #2

leonsteinhaeuser committed:
doc: added pointer types and updated examples

martinrode committed:
json: improved type when using "*json"

We no longer return &interface{} as type when using *json, but instead we maintain the actual type of the JSON (as created by unmarshal) and return a pointer to it.

martinrode committed:
Renamed CSV to CSVParser
